subsys/spm/Kconfig

```diff
@@ -66,6 +76,21 @@ config SPM_SECURE_SERVICES
 	  Note: Please set this and SPM_SERVICE_* configs from the app, instead
 	  of the SPM. This ensures that the values are in sync between the two
 	  images.
+
+config SPM_NS_DEBUG_LOGS_DUMP
```

Comment:
Since you need to enable multiple configurations in the child image SPM when the parent sets this option, you should consider doing it automatically. 

Look at how the overlay file is added here https://github.com/nrfconnect/sdk-nrf/blob/master/samples/CMakeLists.txt#L60

then you can add a check for this option and add the overlay here https://github.com/nrfconnect/sdk-nrf/blob/master/samples/CMakeLists.txt#L21

With this in place a user simply needs to enable option this in their application to have the correct kconfig in the SPM child image.

One missing piece is the manual modification of the DT overlay, but if you improve the error code printed when the wrong console is selected that would be improved as well.

Reply:
done, Added `samples/spm/spm_ns_debug_logs_dump.conf` which is added to spm if `SPM_NS_DEBUG_LOGS_DUMP` is enabled in the app. Also rephrased a bit build assert message.